firefox-ios/Client/Frontend/Autofill/LoginListViewModel.swift

```diff
@@ -0,0 +1,77 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import SwiftUI
+import Storage
+
+@MainActor
+class LoginListViewModel: ObservableObject {
+    @Published var logins: [EncryptedLogin] = []
+
+    private let tabURL: URL
+    private let loginStorage: LoginStorage
+    private let logger: Logger
+    let onLoginCellTap: (EncryptedLogin) -> Void
+    let manageLoginInfoAction: () -> Void
+
+    var shortDisplayString: String {
+        tabURL.baseDomain ?? ""
+    }
+
+    init(
+        tabURL: URL,
+        loginStorage: LoginStorage,
+        logger: Logger,
+        onLoginCellTap: @escaping (EncryptedLogin) -> Void,
+        manageLoginInfoAction: @escaping () -> Void
+    ) {
+        self.tabURL = tabURL
+        self.loginStorage = loginStorage
+        self.logger = logger
+        self.onLoginCellTap = onLoginCellTap
+        self.manageLoginInfoAction = manageLoginInfoAction
+    }
+
+    func fetchLogins() async {
+        do {
+            let logins = try await loginStorage.listLogins()
+            self.logins = logins.filter { login in
+                guard let recordHostnameURL = URL(string: login.hostname) else { return false }
+                return recordHostnameURL.baseDomain == tabURL.baseDomain
+            }
+        } catch {
+            self.logger.log("Error fetching logins",
+                            level: .warning,
+                            category: .address,
+                            description: "Error fetching addresses: \(error.localizedDescription)")
+        }
+    }
+}
+
+class MockLogger: Logger {
```

Comment:
I know we use mock usually for tests but I found them useful for SwiftUI previews for speed of development and compilation.

Reply:
I think this should be in its own file rather than under the LoginListViewModel. If this is a pattern we will follow more (having mocks inside production code) I would have a specific folder for mocks maybe? In this PR there's `MockLogger` and `MockLoginStorage` that is introduced, as part of other files. Will be hard to discover and know they exists at their current location

Reply:
Now thinking, will that have any conflicts with current existing `MockLogger` under the tests target? Should that one exists over there if we have one in the prod code? Feels like duplication.

Reply:
Same as the Storage, we need a way to mock the live implementation with mocks for SwiftUI previews. Having the mocks in test targets will not be accessible in the main target.

Reply:
We discussed in DM and will leave this be for now